str.lisp

```diff
@@ -230,21 +285,15 @@
 (defun insert (string/char index s)
   "Insert the given string (or character) at the `index' into `s' and return a new string.
 
-  If `index' is out of bounds, ignore and return `s'."
-  (when (characterp string/char)
-    (setf string/char (string string/char)))
+   If `index' or `string/char' is NIL, ignore and return `s'."
   (cond
-    ((null index)
-     s)
-    ((< index 0)
-     s)
-    ((> index (length s))
-     s)
-    (t
-     (concatenate 'string
-                  (subseq s 0 index)
-                  string/char
-                  (subseq s index)))))
+    ((and s index string/char)
+     ;; insert _ in "abcd" at -1 means "abcd_", not "abc_d"
+     (with-indices (index) (1+ (length s))
+       (concat (slice 0 index s t)
+               (string string/char)
+               (slice index (length s) s t))))
```

Comment:
use nil/t instead of (length s)

Reply:
Yes, I am thinking about adding some `if-out-of-bounds` keyword argument, we can talk more about it in the issue, if you have questions or preferences.